pyadjoint/reduced_functional.py

```diff
@@ -124,13 +124,15 @@ def derivative(self, adj_input=1.0, options={}):
 
         # Scale adjoint input
         with stop_annotating():
-            adj_value = self.scale * adj_input
+            # Make sure adj_input is an OverloadedType
+            adj_input = create_overloaded_object(adj_input)
+            adj_input._ad_mul(self.scale)
```

Comment:
Split this to a separate PR.

Reply:
Done, see https://github.com/dolfin-adjoint/pyadjoint/pull/112